tests/tests/test-xcm/test-mock-hrmp-transact-ethereum.ts

```diff
@@ -0,0 +1,578 @@
+      statemint_para_id,
+      totalMessage,
+    ]);
+    // Create a block in which the XCM will be executed
+    await context.createBlock();
+    // Make sure descended address has the transferred foreign assets (minus the xcm fees).
+    expect(
+      (await context.polkadotApi.query.assets.account(assetId, descendedAddress))
+        .unwrap()
+        .balance.toBigInt()
+    ).to.eq(assetsToTransfer);
+  });
+
+  it("Should receive transact and should be able to execute", async function () {
+    const xcmTransactions = [
+      {
+        V1: {
+          gas_limit: 100000,
+          fee_payment: {
+            Auto: {
+              Low: null,
+            },
+          },
+          action: {
+            Call: contractDeployed.options.address,
+          },
+          value: 0n,
+          input: contractDeployed.methods.incr().encodeABI().toString(),
+          access_list: null,
+        },
+      },
+      {
+        V2: {
+          gas_limit: 100000,
+          action: {
+            Call: contractDeployed.options.address,
+          },
+          value: 0n,
+          input: contractDeployed.methods.incr().encodeABI().toString(),
+          access_list: null,
+        },
+      },
+    ];
+
+    let expectedCalls = 0n;
+
+    for (const xcmTransaction of xcmTransactions) {
+      expectedCalls++;
+
+      // TODO need to update lookup types for xcm ethereum transaction V2
+      const transferCall = context.polkadotApi.tx.ethereumXcm.transact(xcmTransaction);
+      const transferCallEncoded = transferCall?.method.toHex();
+      // We are going to test that we can receive a transact operation from parachain 1
+      // using descendOrigin first
+      const xcmMessage = {
+        V2: [
+          {
+            DescendOrigin: {
+              X1: {
+                AccountKey20: {
+                  network: "Any",
+                  key: sendingAddress,
+                },
+              },
+            },
+          },
+          {
+            WithdrawAsset: [
+              {
+                id: {
+                  Concrete: ASSET_MULTILOCATION,
+                },
+                fun: { Fungible: assetsToTransfer / 2n },
+              },
+            ],
+          },
+          {
+            BuyExecution: {
+              fees: {
+                id: {
+                  Concrete: ASSET_MULTILOCATION,
+                },
+                fun: { Fungible: assetsToTransfer / 2n },
+              },
+              weightLimit: { Limited: assetsToTransfer / 2n },
+            },
+          },
+          {
+            Transact: {
+              originType: "SovereignAccount",
+              requireWeightAtMost: new BN(2_500_000_000), // 100_000 gas
+              call: {
+                encoded: transferCallEncoded,
+              },
+            },
+          },
+        ],
+      };
+      const xcmpFormat: XcmpMessageFormat = context.polkadotApi.createType(
```

Comment:
I am starting to think we should make a helper funciton send_hrmp, where the XcmVersionedXcm and XcmpMessageFormat are set. Not for this PR though

Reply:
Agreed, probably some set of utilities to construct this messages will help a lot for test readability